From 00011b0208bbf96e7995fc2b7fe0ec23e66ec3bb Mon Sep 17 00:00:00 2001
From: Rocky Hao <rocky.hao@rock-chips.com>
Date: Thu, 26 Apr 2018 16:57:59 +0800
Subject: [PATCH] arm: dts: rockchip: update tsadc's pinctrl for rk3288

we set cru reset as tshut's default mode, i.e. hw-tshut-mode = <0>, and
otp_gpio mode is set accordingly. if gpio mode is set,
i.e. hw-tshut-mode = <1>, otp_out is set accordingly.

Change-Id: I3cb4588fec171e2867a326f55c2115b1da927ac3
Signed-off-by: Rocky Hao <rocky.hao@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts      | 2 --
 arch/arm/boot/dts/rk3288-evb-android-rk818-hdmi.dts     | 2 --
 arch/arm/boot/dts/rk3288-evb-android-rk818-lvds.dts     | 2 --
 arch/arm/boot/dts/rk3288-evb-android-rk818-mipi-edp.dts | 2 --
 arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts     | 2 --
 arch/arm/boot/dts/rk3288-evb-rk1608.dts                 | 1 -
 arch/arm/boot/dts/rk3288-evb.dtsi                       | 1 -
 arch/arm/boot/dts/rk3288-fennec.dts                     | 1 -
 arch/arm/boot/dts/rk3288-firefly-reload-linux.dts       | 2 --
 arch/arm/boot/dts/rk3288-firefly-reload.dts             | 2 --
 arch/arm/boot/dts/rk3288-firefly-rk808.dts              | 2 --
 arch/arm/boot/dts/rk3288-firefly.dtsi                   | 1 -
 arch/arm/boot/dts/rk3288-miniarm.dts                    | 1 +
 arch/arm/boot/dts/rk3288-popmetal-android-vga.dts       | 1 -
 arch/arm/boot/dts/rk3288-popmetal-android.dts           | 1 -
 arch/arm/boot/dts/rk3288-popmetal.dts                   | 1 -
 arch/arm/boot/dts/rk3288-r89.dts                        | 1 -
 arch/arm/boot/dts/rk3288-rock2-som.dtsi                 | 1 -
 arch/arm/boot/dts/rk3288-veyron.dtsi                    | 1 +
 arch/arm/boot/dts/rk3288-x7811-rk818-dual-dsi.dts       | 2 --
 arch/arm/boot/dts/rk3288.dtsi                           | 3 ++-
 21 files changed, 4 insertions(+), 28 deletions(-)

diff --git a/arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts
index ccf8d6bb15e6..2e922ac2ac03 100644
--- a/arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts
@@ -405,9 +405,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-evb-android-rk818-hdmi.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-hdmi.dts
index 424201fd5cab..fdfa2805c19f 100644
--- a/arch/arm/boot/dts/rk3288-evb-android-rk818-hdmi.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-hdmi.dts
@@ -376,9 +376,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-evb-android-rk818-lvds.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-lvds.dts
index cf0ecf55b534..a3cf184eabc6 100644
--- a/arch/arm/boot/dts/rk3288-evb-android-rk818-lvds.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-lvds.dts
@@ -401,9 +401,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi-edp.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi-edp.dts
index a433052a3fa4..76d5dcce6ba5 100644
--- a/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi-edp.dts
@@ -431,9 +431,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts
index 0a90e44fe656..96042e38c61a 100644
--- a/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts
@@ -400,9 +400,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-evb-rk1608.dts b/arch/arm/boot/dts/rk3288-evb-rk1608.dts
index 42fee1b7e109..3fffa1cb380c 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk1608.dts
+++ b/arch/arm/boot/dts/rk3288-evb-rk1608.dts
@@ -755,7 +755,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <0>; /* tshut polarity 0:LOW 1:HIGH */
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-evb.dtsi b/arch/arm/boot/dts/rk3288-evb.dtsi
index 9a64da482781..f2661a93d1fe 100644
--- a/arch/arm/boot/dts/rk3288-evb.dtsi
+++ b/arch/arm/boot/dts/rk3288-evb.dtsi
@@ -390,7 +390,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <0>; /* tshut polarity 0:LOW 1:HIGH */
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-fennec.dts b/arch/arm/boot/dts/rk3288-fennec.dts
index ff9b8e5de403..3e02e5897a60 100644
--- a/arch/arm/boot/dts/rk3288-fennec.dts
+++ b/arch/arm/boot/dts/rk3288-fennec.dts
@@ -462,7 +462,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <0>; /* tshut polarity 0:LOW 1:HIGH */
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts b/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts
index 544f322d29f6..3cefff74d004 100644
--- a/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts
+++ b/arch/arm/boot/dts/rk3288-firefly-reload-linux.dts
@@ -427,9 +427,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-firefly-reload.dts b/arch/arm/boot/dts/rk3288-firefly-reload.dts
index ebf5a65ce8b7..e0ea735ced71 100644
--- a/arch/arm/boot/dts/rk3288-firefly-reload.dts
+++ b/arch/arm/boot/dts/rk3288-firefly-reload.dts
@@ -734,9 +734,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-firefly-rk808.dts b/arch/arm/boot/dts/rk3288-firefly-rk808.dts
index 18dc5770c721..ff4890518851 100644
--- a/arch/arm/boot/dts/rk3288-firefly-rk808.dts
+++ b/arch/arm/boot/dts/rk3288-firefly-rk808.dts
@@ -754,9 +754,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288-firefly.dtsi b/arch/arm/boot/dts/rk3288-firefly.dtsi
index 5ef6eae9d406..b8f6d4364172 100644
--- a/arch/arm/boot/dts/rk3288-firefly.dtsi
+++ b/arch/arm/boot/dts/rk3288-firefly.dtsi
@@ -726,7 +726,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>;
 	rockchip,hw-tshut-polarity = <0>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-miniarm.dts b/arch/arm/boot/dts/rk3288-miniarm.dts
index 8f3c406de0da..a1744fb3a570 100644
--- a/arch/arm/boot/dts/rk3288-miniarm.dts
+++ b/arch/arm/boot/dts/rk3288-miniarm.dts
@@ -597,6 +597,7 @@
 &tsadc {
 	rockchip,hw-tshut-mode = <1>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
+	pinctrl-1 = <&otp_out>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/rk3288-popmetal-android-vga.dts b/arch/arm/boot/dts/rk3288-popmetal-android-vga.dts
index 216e44fb11c0..edb427795c68 100644
--- a/arch/arm/boot/dts/rk3288-popmetal-android-vga.dts
+++ b/arch/arm/boot/dts/rk3288-popmetal-android-vga.dts
@@ -575,7 +575,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>;
 	rockchip,hw-tshut-polarity = <0>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-popmetal-android.dts b/arch/arm/boot/dts/rk3288-popmetal-android.dts
index faac6f5abadd..c57138334635 100644
--- a/arch/arm/boot/dts/rk3288-popmetal-android.dts
+++ b/arch/arm/boot/dts/rk3288-popmetal-android.dts
@@ -552,7 +552,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>;
 	rockchip,hw-tshut-polarity = <0>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-popmetal.dts b/arch/arm/boot/dts/rk3288-popmetal.dts
index 1f918b223d68..f9b023bfe647 100644
--- a/arch/arm/boot/dts/rk3288-popmetal.dts
+++ b/arch/arm/boot/dts/rk3288-popmetal.dts
@@ -494,7 +494,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>;
 	rockchip,hw-tshut-polarity = <0>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-r89.dts b/arch/arm/boot/dts/rk3288-r89.dts
index 312f5ef403f4..9a4f6d66c8fb 100644
--- a/arch/arm/boot/dts/rk3288-r89.dts
+++ b/arch/arm/boot/dts/rk3288-r89.dts
@@ -351,7 +351,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>;
 	rockchip,hw-tshut-polarity = <0>;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-rock2-som.dtsi b/arch/arm/boot/dts/rk3288-rock2-som.dtsi
index 6cd794145b35..247f41571d80 100644
--- a/arch/arm/boot/dts/rk3288-rock2-som.dtsi
+++ b/arch/arm/boot/dts/rk3288-rock2-som.dtsi
@@ -277,7 +277,6 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <0>; /* tshut polarity 0:LOW 1:HIGH */
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/rk3288-veyron.dtsi b/arch/arm/boot/dts/rk3288-veyron.dtsi
index 9069f83169e1..066267bbe878 100644
--- a/arch/arm/boot/dts/rk3288-veyron.dtsi
+++ b/arch/arm/boot/dts/rk3288-veyron.dtsi
@@ -364,6 +364,7 @@
 
 	rockchip,hw-tshut-mode = <1>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
+	pinctrl-1 = <&otp_out>;
 };
 
 &uart0 {
diff --git a/arch/arm/boot/dts/rk3288-x7811-rk818-dual-dsi.dts b/arch/arm/boot/dts/rk3288-x7811-rk818-dual-dsi.dts
index 9cd14b614a02..1c3fb47b1f03 100644
--- a/arch/arm/boot/dts/rk3288-x7811-rk818-dual-dsi.dts
+++ b/arch/arm/boot/dts/rk3288-x7811-rk818-dual-dsi.dts
@@ -382,9 +382,7 @@
 };
 
 &tsadc {
-	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	rockchip,hw-tshut-polarity = <1>; /* tshut polarity 0:LOW 1:HIGH */
-	pinctrl-1 = <&otp_gpio>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 77e592b4f4fd..de94aa450a2b 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -622,10 +622,11 @@
 		reset-names = "tsadc-apb";
 		pinctrl-names = "init", "default", "sleep";
 		pinctrl-0 = <&otp_gpio>;
-		pinctrl-1 = <&otp_out>;
+		pinctrl-1 = <&otp_gpio>;
 		pinctrl-2 = <&otp_gpio>;
 		#thermal-sensor-cells = <1>;
 		rockchip,hw-tshut-temp = <95000>;
+		rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 		status = "disabled";
 	};
 
-- 
2.35.3

